Source/DafnyCore/AST/Grammar/Printer/Printer.Expression.cs

```diff
@@ -1150,9 +1155,14 @@ void PrintExpr(Expression expr, int contextBindingStrength, bool fragileContext,
         bool parensNeeded = !isRightmost;
         if (parensNeeded) { wr.Write("("); }
         wr.Write("if ");
-        PrintExpression(ite.Test, false);
+        if (options.DafnyPrintResolvedFile == null) {
+          PrintGuard(ite.IsBindingGuard, ite.Test);
+        } else {
+          PrintExpression(ite.Test, false);
+        }
         wr.Write(" then ");
-        PrintExpression(ite.Thn, false);
+        var thenBranch = options.DafnyPrintResolvedFile == null && ite.IsBindingGuard ? ((LetExpr)ite.Thn).Body : ite.Thn;
+        PrintExpression(thenBranch, false);
```

Comment:
Do we have tests for the printer?

Reply:
We did (for `PrintExtendedExpr`, but) not for `PrintExpr`. I have now added one.